app/controllers/nonsipp/TaskListController.scala

```diff
@@ -69,7 +69,7 @@ class TaskListController @Inject()(
         } else {
           userAnswersUnchangedAllSections(
             request.userAnswers,
-            if (isSubmitted(response.head)) {
+            if (isSubmitted(response.maxBy(_.reportVersion))) {
```

Comment:
Do we need a test to simulate the wrong order?

Reply:
Do we really need to test maxBy here ? Instead of making assumption head or last, we're always selecting record with the greatest report version. `head` was okay for stub data (it's reverse ordered) in local or staging but etmp sending in ascending order.

Reply:
I meant in "TaskListControllerSpec", do we need a test where a wrong order is mocked out and to check if pureUserAnswers used for comparison.